FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1494 @@
+
+/**
+ * @brief Maximum number of subscriptions to store in the subscription list.
+ */
+#define SUBSCRIPTIONS_MAX_COUNT             10
+
+/**
+ * @brief Number of publishes done by the publisher in this demo.
+ */
+#define PUBLISH_COUNT                       16
+
+/**
+ * @brief Size of statically allocated buffers for holding topic names and payloads in this demo.
+ */
+#define DEMO_BUFFER_SIZE                    100
+
+/**
+ * @brief Size of dynamically allocated buffers for holding topic names and payloads in this demo.
+ */
+#define DYNAMIC_BUFFER_SIZE                 25
+
+/**
+ * @brief Max number of commands that can be enqueued.
+ */
+#define COMMAND_QUEUE_SIZE                  25
+
+/**
+ * @brief Max number of received publishes that can be enqueued for a task.
+ */
+#define PUBLISH_QUEUE_SIZE                  20
+
+/**
+ * @brief Delay for the subscriber task when no publishes are waiting in the queue.
+ */
+#define SUBSCRIBE_TASK_DELAY_MS             400U
+
+/**
+ * @brief Delay for the publisher task between synchronous publishes.
+ */
+#define PUBLISH_DELAY_SYNC_MS               500U
+
+/**
+ * @brief Delay for the publisher task between asynchronous publishes.
+ */
+#define PUBLISH_DELAY_ASYNC_MS              50U
+
+/**
+ * @brief Notification bit indicating completion of publisher task.
+ */
+#define PUBLISHER_TASK_COMPLETE_BIT         ( 1U << 1 )
+
+/**
+ * @brief Notification bit indicating completion of subscriber task.
+ */
+#define SUBSCRIBE_TASK_COMPLETE_BIT         ( 1U << 2 )
+
+/**
+ * @brief Notification bit used by subscriber task for subscribe operation.
+ */
+#define SUBSCRIBE_COMPLETE_BIT              ( 1U << 0 )
+
+/**
+ * @brief Notification bit used by subscriber task for unsubscribe operation.
+ */
+#define UNSUBSCRIBE_COMPLETE_BIT            ( 1U << 1 )
+
+/**
+ * @brief Maximum number of loop iterations to wait for a task notification.
+ */
+#define MAX_WAIT_ITERATIONS                 5
+
+/**
+ * @brief Topic filter used by the subscriber task.
+ */
+#define SUBSCRIBE_TOPIC_FILTER              "publish/+/filter"
+
+/**
+ * @brief Format string used by the publisher task for topic names.
+ */
+#define PUBLISH_TOPIC_FORMAT_STRING         "publish/%i/filter"
+
+/**
+ * @brief Format string used by the publisher task for payloads.
+ */
+#define PUBLISH_PAYLOAD_FORMAT              "Hello World! %d"
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief A type of command for interacting with the MQTT API.
+ */
+typedef enum CommandType
+{
+    PROCESSLOOP, /**< @brief Call MQTT_ProcessLoop(). */
+    PUBLISH,     /**< @brief Call MQTT_Publish(). */
+    SUBSCRIBE,   /**< @brief Call MQTT_Subscribe(). */
+    UNSUBSCRIBE, /**< @brief Call MQTT_Unsubscribe(). */
+    PING,        /**< @brief Call MQTT_Pint(). */
+    DISCONNECT,  /**< @brief Call MQTT_Disconnect(). */
+    CONNECT,     /**< @brief Placeholder command for reconnecting a broken connection. */
```

Comment:
Why is this a placeholder command?

Reply:
Meant to be in a subsequent PR. I didn't want to add anything more to this one

Reply:
Ok Sounds good. Thanks